src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionStringBuilder.cs

```diff
@@ -169,1766 +184,1676 @@ static SqlConnectionStringBuilder()
-        [DisplayName(DbConnectionStringKeywords.InitialCatalog)]
-        [ResCategoryAttribute(StringsHelper.ResourceNames.DataCategory_Source)]
-        [ResDescriptionAttribute(StringsHelper.ResourceNames.DbConnectionString_InitialCatalog)]
-        [RefreshPropertiesAttribute(RefreshProperties.All)]
-        [TypeConverter(typeof(SqlInitialCatalogConverter))]
-        public string InitialCatalog
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnectionStringBuilder.xml' path='docs/members[@name="SqlConnectionStringBuilder"]/ctorConnectionString/*' />
+        public SqlConnectionStringBuilder(string connectionString) : base()
         {
-            get { return _initialCatalog; }
-            set
+            if (!string.IsNullOrEmpty(connectionString))
             {
-                SetValue(DbConnectionStringKeywords.InitialCatalog, value);
-                _initialCatalog = value;
+                ConnectionString = connectionString;
             }
         }
 
-        /// <include file='..\..\..\..\..\..\..\doc\snippets\Microsoft.Data.SqlClient\SqlConnectionStringBuilder.xml' path='docs/members[@name="SqlConnectionStringBuilder"]/IntegratedSecurity/*' />
-        [DisplayName(DbConnectionStringKeywords.IntegratedSecurity)]
-        [ResCategoryAttribute(StringsHelper.ResourceNames.DataCategory_Security)]
-        [ResDescriptionAttribute(StringsHelper.ResourceNames.DbConnectionString_IntegratedSecurity)]
-        [RefreshPropertiesAttribute(RefreshProperties.All)]
-        public bool IntegratedSecurity
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnectionStringBuilder.xml' path='docs/members[@name="SqlConnectionStringBuilder"]/Item/*' />
+        public override object this[string keyword]
         {
-            get { return _integratedSecurity; }
-            set
+            get
             {
-                SetValue(DbConnectionStringKeywords.IntegratedSecurity, value);
-                _integratedSecurity = value;
+                Keywords index = GetIndex(keyword);
+                return GetAt(index);
             }
-        }
-
-        /// <include file='..\..\..\..\..\..\..\doc\snippets\Microsoft.Data.SqlClient\SqlConnectionStringBuilder.xml' path='docs/members[@name="SqlConnectionStringBuilder"]/Authentication/*' />
-        [DisplayName(DbConnectionStringKeywords.Authentication)]
-        [ResCategoryAttribute(StringsHelper.ResourceNames.DataCategory_Security)]
-        [ResDescriptionAttribute(StringsHelper.ResourceNames.DbConnectionString_Authentication)]
-        [RefreshPropertiesAttribute(RefreshProperties.All)]
-        public SqlAuthenticationMethod Authentication
-        {
-            get { return _authentication; }
             set
             {
-                if (!DbConnectionStringBuilderUtil.IsValidAuthenticationTypeValue(value))
+                if (null != value)
```

Comment:
`value is not null` or `value != null` ?

Reply:
The new one is an improvement, I just prefer the c# idomatic way of using variable == const rather than the C way. 😁